### PR TITLE
Add HAL driver to talk to P260C I/O boards on an RS-485 chain.

### PR DESCRIPTION
**Pull request by ArcEye**
Author Colin Whittacker

Signed-off-by: Mick <arceye@mgware.co.uk>

### CODE REVIEW

File: src/hal/drivers/hal_p260c.c [hunk 1]
```diff
@@ -0,0 +1,1208 @@
...
+	mstat->min_tx_boards = 6;
+	mstat->max_rx_wait = 5000000;
+	mstat->debug_on_error = 0;
+
+	// Debug
+	util_bb_init();
+	configure_output( 915 );
+	configure_output( 923 );
+//	gpio48 = openGPIO( 48 );
+//	gpio49 = openGPIO( 49 );
```
Comment (machinekoder) [inline src/hal/drivers/hal_p260c.c]: dead code

File: src/hal/drivers/hal_p260c.c [hunk 2]
```diff
@@ -0,0 +1,1208 @@
...
+	closeserial( sfd );
+#endif
+
+	hal_exit(comp_id);
+}
+
+/***********************************************************************
+Implementation
+*/
+#if 0 // Can't use GPIO for TX-enable (too slow)
```
Comment (machinekoder) [inline src/hal/drivers/hal_p260c.c]: dead code

File: src/hal/drivers/hal_p260c.c [hunk 3]
```diff
@@ -0,0 +1,1208 @@
...
+//	options.c_cflag &= ~PARENB;
+	options.c_cflag |= PARENB;
+	options.c_cflag &= ~CSTOPB;
+
+	options.c_lflag &= ~(ISIG|ICANON|ECHO|ECHOE|ECHOK|ECHONL|NOFLSH|TOSTOP|ECHOCTL|ECHOKE|FLUSHO);
+
+	// set the read and write speed to
+	// All speeds can be prefixed with B
+	// as a settings.
+//	cfsetispeed(&options, B115200);
```
Comment (machinekoder) [inline src/hal/drivers/hal_p260c.c]: dead code

File: src/hal/drivers/hal_p260c.c [hunk 4]
```diff
@@ -0,0 +1,1208 @@
...
+    if (tcflush(fd, TCIOFLUSH) == -1) {
+       	rtapi_print_msg(RTAPI_MSG_ERR,"openserial(): tcflush()");
+        return -1;
+    }
+    if (tcsetattr(fd, TCSANOW, &options) == -1) {
+       	rtapi_print_msg(RTAPI_MSG_ERR,"initserial(): tcsetattr()");
+        return -1;
+    }
+
+#if 0    // RS485 settings.
```
Comment (machinekoder) [inline src/hal/drivers/hal_p260c.c]: dead code

File: src/hal/drivers/hal_p260c.c [hunk 5]
```diff
@@ -0,0 +1,1208 @@
...
+				*bits = (input_data[1]<<8) & 0xFF00;
+				*bits |= (input_data[2] & 0xFF);
+
+				return input_address;
+			}
+		}
+	}
+	return -1;
+}
+#if 0
```
Comment (machinekoder) [inline src/hal/drivers/hal_p260c.c]: dead code

File: src/hal/drivers/hal_p260c.c [hunk 6]
```diff
@@ -0,0 +1,1208 @@
...
+	boards[board].invalid_timer++;
+
+	return ret;
+}
+//
+// Read for a maximum of 8 bytes time and timeout
+// 8 bytes at 3,000,000 baud is about 30 usec
+// wait for 1/2 msec since the usb turn around is long.
+//
+#if 0
```
Comment (machinekoder) [inline src/hal/drivers/hal_p260c.c]: dead code

**Comment by pmcstone** [comment]
Just wanted to add driver uses hardware from FTDI, USB to RS485

http://www.ftdichip.com/Products/Cables/USBRS485.htm

**Comment by machinekoder** [comment]
@pmcstone what seems more reasonable for you?

**Comment by pmcstone** [comment]
Please clean up accordingly. Thanks!

**Comment by ArcEye** [comment]
I left in the GPIO enable to give some future reader a clue how it is done.
You could remove far more, but it works.

**Comment by machinekoder** [comment]
excellent, thanks

**Comment by pmcstone** [comment]
Can you add company brand?

Author: Colin Whittacker
For: PMC Stone, Inc.

Thanks!

**Comment by ArcEye** [comment]
If you go to https://github.com/machinekit/machinekit/blob/master/src/hal/drivers/hal_p260c.c#L36
Press the edit icon at the top (pencil), change the wording to suit and save it as a PR and one of us can merge it for you

**Pull request merged**